Fix humantime regex error message

drammock opened the pull request:
This PR:

- allows uppercase letters in `humantime` (am/pm)
- makes minutes optional (allows "9am-5:30pm" instead of forcing "9:00am-5:30pm")
- improves the error message when humantime fails to match the regex

drammock committed:
more informative error message

fmichonneau commented:
We use this data for our workshop database (AMY), and so I'm pretty sure we need to enforce these formats. Having a more informative error message would certainly be useful though!

drammock commented:
OK, I'll revert the regex change but keep the error message. stand by.

drammock commented:
OK @fmichonneau I've rebased to remove the changes to the regex and test, this now only affects the error message (which I've updated to not include "9am-5pm" as acceptable).

fmichonneau commented:
Great, thank you!
